eta-unit-testing-project-2019.2-Recife-main/src/test/java/PersonTest.java

```diff
@@ -64,21 +65,33 @@ public void getName_noFirstNameNorLastName_throwsException() {
     }
 
     @Test
-    public void isBirthdayToday_differentMonthAndDay_false() {
-       person.setBirthday(LocalDate.parse("2020-01-01"));
-       boolean result = person.isBirthdayToday();
-       Assertions.assertFalse(result);
+    public void isBirthdayToday_differentMonthAndDay_false() {     
+        PersonService service = new PersonService();
+        person.setBirthday(service.getNow());
+        boolean result = person.isBirthdayToday();
+        Assertions.assertFalse(result);
        
     }
 
     @Test
     public void isBirthdayToday_sameMonthDifferentDay_false() {
-    	Date date = new Date();
-    	LocalDate localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
-    	int year = localDate.getYear();
-    	int month = localDate.getMonthValue();
-    	int day = localDate.getDayOfMonth();
-    	person.setBirthday(LocalDate.of(year, month, (day -2)));
+        person = new Person(){
+            
+            @Override
+            public LocalDate getNow()
+            {  
+                Date date = new Date();
+                LocalDate localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+                int year = localDate.getYear();
+                int month = localDate.getMonthValue();
+                int diferentDay = localDate.getDayOfMonth();
+                return LocalDate.of(year, month, diferentDay-5);
```

Comment:
Além de deixar o entendimento do teste difícil, o cálculo `diferentDay-5` pode resultar em valores que variam de -4 até 0, quando o dia é igual ou inferior a 5, o que resulta em `DateTimeException`.

https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html#of-int-int-int-

Por que não utilizar uma data fixa aqui?